test/core.js

```diff
@@ -67,6 +67,16 @@ test( "attribute selectors with naked '#'", function() {
 			jQuery( "a[href=\"#junk\"]" );
 		} catch ( e ) {}
 	} );
+
+	expectNoWarning( "ready function with attribute selector", function() {
+		try {
+			jQuery( function() {
+				if ( jQuery.thisIsNeverTrue ) {
+					jQuery( "a[href=#]" );
```

Comment:
Since the code never reaches this statement, what's its purpose? I don't understand what's exactly tested here.